client/src/app/shared-components/recipe-search-and-select/recipe-search-and-select.js

```diff
@@ -0,0 +1,82 @@
+import { useRef, useState } from "react";
+import InputField from "../../../shared/input/input-component";
+import { apiKey } from "../../../config/cooking-apiKey";
+import Spinner from "react-bootstrap/esm/Spinner";
+import "./recipe-search-and-select-style.scss";
+
+const RecipeSearchAndSelect = () => {
+  const [searchValue, setSearchValue] = useState("");
+  const [loadingRecipes, setLoadingRecipes] = useState(false);
+  const [toggleDropDown, setToggleDropdown] = useState(false);
+  const [recipeOptions, setRecipeOptions] = useState([]);
+  const [selectedRecipe, setSelectedRecipe] = useState({});
+  const timeout = useRef();
+  let url = `https://api.spoonacular.com/recipes/complexSearch?apiKey=${apiKey}&query=${searchValue}`;
+  const handleChange = (value) => {
+    setToggleDropdown(true);
+    setLoadingRecipes(true);
+    setSearchValue(value);
+    if (!value) {
```

Comment:
Let's do this check at the top of the handleChange()

Reply:
Rumyana: Cannot move at the top itself as that creates bugs. Moved on top of other checks that were higher up the code execution chain earlier.

Reply:
Bancho: OK, good enough for now. Will create a note-to-self and possibly a separate issue to investigate said bugs and if further tech-wise changes are required.